recipe/build.sh

```diff
@@ -5,134 +5,104 @@
+        -no-avx \
+        -D _X_INLINE=inline \
+        -D XK_dead_currency=0xfe6f \
+        -D XK_ISO_Level5_Lock=0xfe13 \
+        -D FC_WEIGHT_EXTRABLACK=215 \
+        -D FC_WEIGHT_ULTRABLACK=FC_WEIGHT_EXTRABLACK \
+        -D GLX_GLXEXT_PROTOTYPES"
+
+  # To get a much quicker turnaround you can add this: (remember also to add the backslash after GLX_GLXEXT_PROTOTYPES)
+  # -skip qtwebsockets -skip qtwebchannel -skip qtwayland -skip qtsvg -skip qtsensors -skip qtcanvas3d -skip qtconnectivity -skip declarative -skip multimedia -skip qttools
+
+  # If we must not remove strict_c++ from qtbase/mkspecs/features/qt_common.prf
+  # (0007-qtbase-CentOS5-Do-not-use-strict_c++.patch) then we need to add these
+  # defines instead:
+  # -D __u64="unsigned long long" \
+  # -D __s64="__signed__ long long" \
+  # -D __le64="unsigned long long" \
+  # -D __be64="__signed__ long long"
+elif [ $(uname) == Darwin ]; then
+  # Let Qt set its own flags and vars
+  for x in OSX_ARCH CFLAGS CXXFLAGS LDFLAGS; do
+    unset $x
+  done
+  export MACOSX_DEPLOYMENT_TARGET=10.7
+  OPTS="-no-egl \
+        -no-openssl \
+        -qt-freetype \
+        -qt-libjpeg \
+        -qt-libpng \
+        -c++11 \
+        -no-framework \
+        -no-dbus \
+        -no-mtdev \
+        -no-harfbuzz \
+        -no-xinput2 \
+        -no-xcb-xlib"
 fi
 
-if [ $(uname) == Darwin ]; then
-    # Let Qt set its own flags and vars
-    for x in OSX_ARCH CFLAGS CXXFLAGS LDFLAGS
-    do
-        unset $x
-    done
-
-    export MACOSX_DEPLOYMENT_TARGET=10.7
-    MAKE_JOBS=$(sysctl -n hw.ncpu)
-
-    ./configure -prefix $PREFIX \
-                -libdir $PREFIX/lib \
-                -bindir $PREFIX/bin \
-                -headerdir $PREFIX/include/qt \
-                -archdatadir $PREFIX \
-                -datadir $PREFIX \
-                -L $PREFIX/lib \
-                -I $PREFIX/include \
-                -release \
-                -opensource \
-                -confirm-license \
-                -shared \
-                -nomake examples \
-                -nomake tests \
-                -verbose \
-                -skip enginio \
-                -skip location \
-                -skip sensors \
-                -skip serialport \
-                -skip serialbus \
-                -skip quickcontrols2 \
-                -skip wayland \
-                -skip canvas3d \
-                -skip 3d \
-                -system-zlib \
-                -qt-pcre \
-                -qt-freetype \
-                -qt-libjpeg \
-                -qt-libpng \
-                -c++11 \
-                -no-framework \
-                -no-dbus \
-                -no-mtdev \
-                -no-harfbuzz \
-                -no-xinput2 \
-                -no-xcb-xlib \
-                -no-libudev \
-                -no-egl \
-                -no-openssl
-
-    DYLD_FALLBACK_LIBRARY_PATH=$PREFIX/lib make -j $MAKE_JOBS || exit 1
-    make install
+./configure -prefix $PREFIX \
```

Comment:
I really find it more confusing and prefer to leave it as it was (I was the one who originally created this file and wanted to clearly see what options are used in each OS separately).

Building Qt is a really complex process to add more complexity to it just save 40 lines of code.